src/identity/components/WithAvatarBadge.tsx

```diff
@@ -0,0 +1,58 @@
+import type { Address, Chain } from 'viem';
+import { useAttestation } from '../hooks/useAttestation';
+import { Badge } from './Badge';
+
+type WithAvatarBadgeInnerProps = {
+  children: React.ReactNode;
+  address: Address;
+};
+
+function WithAvatarBadgeInner({ children, address }: WithAvatarBadgeInnerProps) {
+  const attestation = useAttestation(address);
```

Comment:
I like `useAttestation` a lot, athought I am a bit confused on how it works. Let's revisit this in the next PR.